src/librustc_middle/mir/mod.rs

```diff
@@ -2004,6 +2004,15 @@ impl<'tcx> Debug for Operand<'tcx> {
 }
 
 impl<'tcx> Operand<'tcx> {
+    /// Creates a `()` operand.
+    pub fn unit(tcx: TyCtxt<'tcx>, span: Span) -> Self {
```

Comment:
I'd appreciate avoiding new inherent methods on MIR data types that refer to the TyCtxt.

Reply:
Hm, I guess we could add a unit constant as a field to the building context instead :thinking: 

cc @eddyb

Reply:
Or keep the old `unit_rvalue` method and rename it to `unit_operand`.

Reply:
> I'd appreciate avoiding new inherent methods on MIR data types that refer to the TyCtxt.

I didn't actually think about why we would want this.
Afaict the goal is for `librustc_mir` to not depend on `librustc_middle` @Centril?
Will impl it using a free method.

Reply:
@eddyb we need this function in https://github.com/rust-lang/rust/blob/42abbd8878d3b67238f3611b0587c704ba94f39c/src/librustc_mir/transform/promote_consts.rs#L838
So we can't just place it in `librustc_mir_build` :cry:

Reply:
More precisely, the goal is to separate out the MIR data types to its own crate.

Reply:
@lcnr The method would be used only where `unit_rvalue` is used today.
For the other two places, just copy-paste, it's tiny anyway.

Reply:
This function is only used once :man_shrugging: just removed it entirely